src/org/rust/lang/annotator/RustAnnotator.kt

```diff
@@ -29,6 +28,26 @@ public class RustAnnotator : Annotator {
             is RustTypeParam -> {
                 addTextAttributes(element, holder, RustColors.TYPE_PARAMETER)
             }
+            is RustPatIdent -> {
+                if (element.bindingMode?.mut != null) {
+                    addTextAttributes(element.identifier, holder, RustColors.MUT_BINDING)
+                }
+            }
+            is RustPathPart -> {
+                val resolved = RustResolveEngine().resolve(element).resolved
+                when (resolved) {
+                    is RustPatField -> {
+                        if (resolved.bindingMode?.mut != null) {
+                            addTextAttributes(element, holder, RustColors.MUT_BINDING)
+                        }
+                    }
+                    is RustPatIdent -> {
+                        if (resolved.bindingMode?.mut != null) {
+                            addTextAttributes(element, holder, RustColors.MUT_BINDING)
+                        }
+                    }
+                }
```

Comment:
Sorry, not `|` -- `,` should separated conditions.

https://kotlinlang.org/docs/reference/grammar.html#when